bundles/org.openhab.automation.jythonscripting/src/main/java/org/openhab/automation/jythonscripting/JythonScriptEngineFactory.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.jythonscripting;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import javax.script.ScriptEngine;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.OpenHAB;
+import org.openhab.core.automation.module.script.AbstractScriptEngineFactory;
+import org.openhab.core.automation.module.script.ScriptEngineFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * This is an implementation of {@link ScriptEngineFactory} for Jython.
+ *
+ * @author Scott Rushworth - Initial contribution
+ * @author Wouter Born - Initial contribution
+ */
+@Component(service = ScriptEngineFactory.class)
+@NonNullByDefault
+public class JythonScriptEngineFactory extends AbstractScriptEngineFactory {
+
+    private static final String PYTHON_CACHEDIR = "python.cachedir";
+    private static final String PYTHON_HOME = "python.home";
+    private static final String PYTHON_PATH = "python.path";
+
+    private static final String DEFAULT_PYTHON_PATH = Paths
+            .get(OpenHAB.getConfigFolder(), "automation", "lib", "python").toString();
+
+    private static final String SCRIPT_TYPE = "py";
+    private static final javax.script.ScriptEngineManager ENGINE_MANAGER = new javax.script.ScriptEngineManager();
```

Comment:
See https://github.com/openhab/openhab-addons/pull/7208#discussion_r426598405

Reply:
Thanks for testing! The issue is that the `org.openhab.core.automation.module.script.ScriptEngineFactory.ENGINE_MANAGER` has a different classloader. So it only detects engines that are on the boot classpath (like Nashorn). I ran into the same issue with the Groovy add-on. I looked into ways to get engines registered in the `ScriptEngineFactory.ENGINE_MANAGER`. IIRC it wasn't possible to call methods on `ScriptEngineFactory.ENGINE_MANAGER` to (un)register engines provided by OSGi bundles.

Reply:
> Do you have any concerns regarding `SecurityException` which might be thrown by this method?

We don't seem to handle any `SecurityException`s in openhab-core either when modifying System properties:

https://github.com/openhab/openhab-core/search?q=System.setProperty

Reply:
Do we need to handle it?

Reply:
Ah, okay. Sounds reasonable. Thanks for clarification.

> Do we need to handle it?

I think it is okay to not handle it. If the exceptions will be thrown there might be a bigger issue regarding permissions and other stuff will fail too.